samples/apps/copilot-chat-app/webapp/src/redux/features/plugins/pluginsSlice.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Buffer } from 'buffer';
+import { EnablePluginPayload, initialState, Plugin, Plugins, PluginsState } from './PluginsState';
+
+export const pluginsState = createSlice({
+    name: 'plugins',
+    initialState,
+    reducers: {
+        connectPlugin: (state: PluginsState, action: PayloadAction<EnablePluginPayload>) => {
+            var plugin: Plugin;
+            var authData = action.payload.accessToken;
+
+            switch (action.payload.plugin) {
+                case Plugins.MsGraph:
+                    plugin = state.MsGraph;
+                    break;
+                case Plugins.Jira:
+                    plugin = state.Jira;
+                    const encodedData = Buffer.from(
```

Comment:
Curious to learn what Buffer helps here. If it's just base64 encoding, can't we use the native [btoa()](https://developer.mozilla.org/en-US/docs/Web/API/btoa) API in JavaScript?

Reply:
Can we also just pass the `${action.payload.username}:${action.payload.accessToken}` as just a string and not convert to base64 here. We can do that in the back end with the BasicAuthenticationProvider. That will make sure we are converting the string to a UTF8 encoded byte stream before converting to base64. It is also how we are telling users to perform Basic Authentication in the dotnet examples we have.

Reply:
![image](https://user-images.githubusercontent.com/125500434/235500386-08fed934-5d41-4d91-ba5d-b5f6071d306b.png)
btoa has been deprecated and not recommend for use in new applications

Reply:
On your note @amsacha, I'll leave you to change the header as you see fit with the Jira integration. This will help keep the PR scope limited correctly.